### PR TITLE
Fix intermittently failing test

### PR DESCRIPTION
**Pull request by matthew-white**
This is a test of the new form modal:

https://github.com/opendatakit/central-frontend/blob/cc6a098364c637fd06936cf2989e51ea9ecc4fec/lib/components/form/new.vue

### CODE REVIEW

File: test/components/form/new.spec.js [hunk 1]
```diff
@@ -96,15 +96,28 @@ describe('FormNew', () => {
...
+      it('clicking create button during file read shows info message', () => {
+        let originalFunction = null;
+        return Promise.resolve(mountAndMark(FormNew))
           .then(createForm)
+          .then(modal => {
+            originalFunction = FileReader.prototype.readAsText;
+            FileReader.prototype.readAsText = function readAsText() {
+              this.onloadstart();
+            };
+            return modal;
```
[the remark this answers is not in the document]
Comment (matthew-white) [reply test/components/form/new.spec.js]: I can definitely try to run with `const`-ing `originalFunction`.

I can see how modifying the prototype feels a little dangerous... At this point, I haven't come up with any alternatives. I read about modifying `FileReader.prototype` for testing at http://nerds.intuo.io/2016/05/12/mocking-file-uploads-in-javascript.html, though I modified the prototype directly rather than using sinon to do so. Do you think modifying `FileReader.prototype` is reasonably safe in this case? Alternatively, this test is the only time we have a pattern like this, and it’s a test of a bit of an edge case anyway, so part of me feels like another option that wouldn’t be unreasonable would be just to remove the test.

[the remark this answers is not in the document]
Comment (matthew-white) [reply test/components/form/new.spec.js]: I discussed this test with @lognaturel earlier today. I'm going to try to implement slightly different functionality here. Right now, as the file is being read, if you click the Create button, you see an info alert. Instead, I'm going to change it so that as the file is being read, the Create button is disabled. Since the test won't require a click of the Create button, hopefully the file read will still be in progress when the assertion is made, and it won't be necessary to modify the `FileReader` prototype. (There's a chance though that the file read will still be too quick, and we'll have to return to some of these questions about modifying the prototype.) I'll let you know how it goes!

[the remark this answers is not in the document]
Comment (matthew-white) [reply test/components/form/new.spec.js]: FYI, since they both touch alerts, I'm thinking of waiting until we merge #104 before updating this PR.

**Comment by matthew-white** [comment]
I reverted the first commit rather than using `git reset` so that we continue to have access to the commit in case we ever want to return to that approach.
